nav-app/src/app/data.service.ts

```diff
@@ -327,7 +327,10 @@ export abstract class BaseStix {
         this.id = stixSDO.id;
         this.name = stixSDO.name;
         this.description = stixSDO.description;
-        this.attackID = stixSDO.external_references[0].external_id;
+        if (stixSDO.external_references && stixSDO.external_references[0]) this.attackID = stixSDO.external_references[0].external_id; else {
```

Comment:
This won't catch the case where the object has an external_reference but it does not contain the ATT&CK ID. The if statement also needs to check if `external_id` is in the first external reference object.

For reference here's the code we use to deserialize ATT&CK IDs in ATT&CK Workbench: https://github.com/center-for-threat-informed-defense/attack-workbench-frontend/blob/develop/app/src/app/classes/stix/stix-object.ts#L222-L225